ui/components/app/update-modal/update-modal.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import browser from 'webextension-polyfill';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalBody,
+  ModalFooter,
+  Box,
+  Text,
+  ModalHeader,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  BorderRadius,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { setUpdateModalLastDismissedAt } from '../../../store/actions';
+
+function UpdateModal() {
+  const t = useI18nContext();
+
+  return (
+    <Modal
+      isOpen={true}
+      onClose={async () => await setUpdateModalLastDismissedAt(Date.now())}
+      isClosedOnOutsideClick={false}
+      isClosedOnEscapeKey={false}
+      data-testid="update-modal"
+      autoFocus={false}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          onClose={async () => await setUpdateModalLastDismissedAt(Date.now())}
+          startAccessory={true}
+          closeButtonProps={{ 'data-testid': 'update-modal-close-button' }}
+        />
+        <ModalBody display={Display.Flex} flexDirection={FlexDirection.Column}>
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.center}
+            borderRadius={BorderRadius.SM}
+            padding={10}
+          >
+            <img src="/images/logo/metamask-fox.svg" width={160} height={160} />
+          </Box>
+          <Text variant={TextVariant.headingMd} textAlign={TextAlign.Center}>
+            {t('getTheNewestFeatures')}
+          </Text>
+          <Text
+            variant={TextVariant.bodyMd}
+            textAlign={TextAlign.Center}
+            padding={4}
+            paddingBottom={12}
+          >
+            {t('updateInformation')}
+          </Text>
+        </ModalBody>
+        <ModalFooter
+          onSubmit={async () => {
+            await browser.tabs.create({
+              url: 'https://metamask.io/updating',
+              active: true,
+            });
+            browser.runtime.reload();
```

Comment:
Oh noes.

So if a user imports a private key and then within 1000 milliseconds gets the update prompt and clicks the update button the private key will NOT be persisted; _guaranteed_.

😱

Even if we wait for 1 second, or 10 seconds, we don't have any guarantees that the users data will be persisted.

 @Gudahtt Is this a sev1 waiting to happen? (sorry I keep pinging you!)